pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -128,6 +128,10 @@ type ClusterDeploymentSpec struct {
 	// time since the cluster last came out of hibernation.
 	// +optional
 	HibernateAfter *metav1.Duration `json:"hibernateAfter,omitempty"`
+
+	// LimitInstallRestarts is the number of times Hive attempts to restart an installation. If the last attempt
+	// fails ClusterDeployment will show an error in the status
+	LimitInstallRestarts *int `json:"limitInstallRestarts,omitempty"`
```

Comment:
The size of the integer must be specified.
```suggestion
	LimitInstallRestarts *int32 `json:"limitInstallRestarts,omitempty"`
```

Reply:
I kept it `*int` because type of InstallRestarts field in the ClusterDeployment is `*int`. If we change that will we have any impact on existing CRs?

Reply:
You are right that we cannot change the existing `InstallRestarts` type. However, that is not a good enough reason to use the improper `*int` here as well.